marklogic-data-hub/src/main/java/com/marklogic/hub/impl/SimpleHubConfig.java

```diff
@@ -0,0 +1,183 @@
+package com.marklogic.hub.impl;
+
+import com.marklogic.appdeployer.AppConfig;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.HubProject;
+import com.marklogic.hub.step.StepDefinition;
+import org.springframework.util.Assert;
+
+import java.nio.file.Path;
+
+/**
+ * Implementation of HubConfig that does not have a hard dependency on a HubProject. If a HubProject is not set, then
+ * any HubConfig method that depends on a HubProject being present will throw an exception.
+ */
+public class SimpleHubConfig extends AbstractHubConfig {
+
+    private HubProject hubProject;
+
+    public SimpleHubConfig() {
+    }
+
+    public SimpleHubConfig(HubProject hubProject) {
+        this.hubProject = hubProject;
+    }
+
+    /**
+     * Provides a minimally-configured instance of SimpleHubConfig that is suitable for e.g. running a flow.
+     *
+     * @param host
+     * @param mlUsername
+     * @param mlPassword
+     */
+    public SimpleHubConfig(String host, String mlUsername, String mlPassword) {
+        AppConfig appConfig = new AppConfig();
```

Comment:
SimpleHubConfig by design will work only with basic and digest authentication schemes and is not expected to work with SSL / certificate based authentication ?

Reply:
This is just a convenience constructor - you can use the no-arg constructor and set any inputs you want.

But a dirty secret of DHF is that if you want to run a collector for any step, you must use basic/digest authentication - that's because CollectorImpl uses a RestTemplate to talk to the collector and endpoint and only supports username/password.

Reply:
Collector does support SSL/cert auth. We have those tests running in pipeline. It gets the context from client object (which is hubConfig.newStagingClient())
https://github.com/marklogic/marklogic-data-hub/blob/develop/marklogic-data-hub/src/main/java/com/marklogic/hub/collector/impl/CollectorImpl.java#L150

Reply:
SSL is separate from the authentication strategy, and SimpleHubConfig doesn't have any impact on that. "Certificate authentication" is a different kind of authentication that DatabaseClientFactory supports which does not require a username/password. CollectorImpl requires a username/password for authentication, and SimpleHubConfig doesn't have any impact on that.